### ISSUE TITLE
Issue while deployment in polygon_testnet.

### ISSUE THREAD

**Issue by racetrack0616** [truffle-box/polygon-box#4]
![image](https://user-images.githubusercontent.com/68948218/128973813-db543c69-f099-4efa-8781-2c4e0d6e6a67.png)
How to resolve the issue?

**Comment by racetrack0616** [comment]
Sometimes also this
![image](https://user-images.githubusercontent.com/68948218/128973909-9e704e92-bc29-434f-95a0-afd4239d42d8.png)
Totally frustrated with all of these issues. :(

**Comment by fainashalts** [comment]
Hi @racetrack0616, I sympathize -- it can be hard to get up and running on something so new. The documentation and options can be a bit much. I just tested out the `polygon_infura_testnet` and it worked as expected. I am not sure what may be happening with your code. When you use the Polygon Box as-is (running `truffle unbox polygon` and adding your infura key/mnemonic to the `.env` file), are you able to compile and migrate?

If you'd like to share your code, I can try to reproduce on my end. Thanks!

**Comment by fainashalts** [comment]
Oh @racetrack0616 I just had another thought for the Infura endpoint side of things -- you may not have enabled the Polygon network on your Infura account. The Polygon add-on is free, but you may need to indicate that you wish to use it. I think you do so here: https://infura.io/payment?chosenAddon=ethereum_polygon_addon and full documentation for Infura add-ons can be found here: https://infura.io/docs/ethereum#section/Network-Add-Ons

And here is the Infura blog post re: Polygon specifically -- https://blog.infura.io/polygon-now-available/

Please let me know if that helps!

**Comment by racetrack0616** [comment]
Thanks for your response. I haven't tried **polygon_infura_testnet**. I am specially talking about **polygon_testnet**. I saw your comment on the truffle github repo. It would be very much productive if you update the repo with workable configs & especially workable **providerOrUrl**. FYI: I tried **alchemy** url it is also not working. I don't own a card detail right now so unable to test the **infura** option. Seems like there is no any option besides **infura_testnet** cause nothing is actually working. I am not sure if **infura** will work or not. Let's hope that works. The documentation in **polygon** and the repos are misleading which needs immediate update.

**Comment by racetrack0616** [comment]
For now, I am into conclusion that nothing except **infura_testnet** works well on the basis of your comments. But keep this issue open until configs get updated and legit response on other rpc url not working properly.

**Comment by racetrack0616** [comment]
Give yourself a try deploying via **remix** and other rpc urls than **infura_testnet**. It worked for me. Doesn't make sense why's not working with truffle.

**Comment by fainashalts** [comment]
@racetrack0616 thank you for giving me more data on this. I will try to look into it this week and see what I can do to sort it out. Can you try setting `gas: 0` in your Truffle Polygon config? This will address a known issue we'll have a fix for soon, and may help. Please let me know if it works as that may help guide me in figuring this out. Thanks so much!

**Comment by ethernetweb** [comment]
> @racetrack0616 Can you try setting `gas: 0` in your Truffle Polygon config?

Tried this. Same error. Does not have any effect. Thanks for checking into it. The one thing that I've been able to determine on my own, is that it isn't necessarily a TRUFFLE issue as much as it is an RPC issue that occurred after EIP1559. HARDHAT stopped working at the same time.

**Comment by racetrack0616** [comment]
If so then, what is the valid RPC url? And why is **remix** working and not the truffle and hardhat? Why can't any one tell the working RPC url with confidence?

**Comment by omar-saadoun** [comment]
I'm facing the same issue, tried with all RPC URLs getting different errors, not found, too many requests and the most positive one is about never ending confirmations primarily with infura or alchemy, I just deploy with Remix with no issues but truffle never worked to me previously to london:

Starting migrations...
======================
> Network name:    'polygon_testnet'
> Network id:      80001
> Block gas limit: 20000000 (0x1312d00)


1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------
   > transaction hash:    0x603dc8adf79aaadf4d0a349710b974a432470b22ec3f11e4220addb9830e6f81
 Blocks: 43           Seconds: 88


Starting migrations...
======================
> Network name:    'polygon_testnet'
> Network id:      80001
> Block gas limit: 20000000 (0x1312d00)


1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------
   > transaction hash:    0x603dc8adf79aaadf4d0a349710b974a432470b22ec3f11e4220addb9830e6f81
   ⠦ Blocks: 319          Seconds: 656^C%

**Comment by racetrack0616** [comment]
Totally frustrating. It has already been so long I inquire about this issue but nothing seems to be legit

**Comment by fainashalts** [comment]
Hi @racetrack0616 the networks and documentation have been updated to reflect the Infura networks, which I have tested and confirm are working as expected. I hope you have been able to deploy your project without further issue. I'm closing this issue as I believe the problem has been resolved, but please feel free to open a new issue if you feel I have missed something. Thank you for bringing this to my attention!

**Comment by PxGnome** [comment]
> Hi @racetrack0616 the networks and documentation have been updated to reflect the Infura networks, which I have tested and confirm are working as expected. I hope you have been able to deploy your project without further issue. I'm closing this issue as I believe the problem has been resolved, but please feel free to open a new issue if you feel I have missed something. Thank you for bringing this to my attention!



Hi - seeing an issue with RPCs and mainnet similar to the issue described here. However we are able to get it working for Mumbai testnet so not sure what the issue is other than some RPC issue? We've tried all the public RPC and infrua

**Comment by dkillen** [comment]
Hi @racetrack0616 

I have been trying to reproduce your issue but, have been unable to. I have tested with the same environment as per your inital report of this issue [here](https://github.com/trufflesuite/truffle/issues/4240):

truffle v5.1.65
node v12.18.3
npm v6.14.6

I have tested with the following public rpc endoints:

https://rpc-mumbai.maticvigil.com
https://matic-mumbai.chainstacklabs.com
https://matic-testnet-archive-rpc.bwarelabs.com
wss://matic-testnet-archive-ws.bwarelabs.com

I have also set up an Infura and Alchemy rpc endpoint for Polygon for testing:

https://polygon-mumbai.infura.io/v3/<INFURA_PROJECT_ID>
https://polygon-mumbai.g.alchemy.com/v2/<ALCHEMY_PROJECT_ID>

I was able to successfully deploy to the Polygon testnet with each endpoint, except for https://matic-testnet-archive-rpc.bwarelabs.com which returned a 503 error. Given that all other https endpoints work and the bwarelabs wss rpc works, that strongly suggests that the bwarelabs webserver was unavailable at the time of the tests.

Could you please provide us with a link to your GitHub repo if you have one. Alternatively, could you provide us with the whole of your truffle configuration files used in your project. Just remember to remove any unique project id or mnemonic that might be present in your configuration before you do. If we are able to inspect your configuration in full, we may be able to identify the source of your problem.

**Comment by kiknaio** [comment]
I have the exact same issue. Deployment on the Mumbai network works without any problem, but when I try to deploy the same contract on Mainnet, I'm facing an issue.
I've tried Alchemy as well as Infura RPC, but still no result.

**Comment by racetrack0616** [comment]
Thank you @dkillen  for putting on so much effort to reproduce. I somehow managed to deploy in **Polygon mumbai** but failed in **Polygon mainnet**. Can you please help to figure out the issue in Mainnet deployment?

**Comment by PxGnome** [comment]
I second the comments above to deploying to mainnet (no issue on mumbai for me)

Suspect there are some issues with the mainnet as my code works fine at times and at sometimes I won't get anything (not even a txn hash) and then a few hours later all my pending transactions will suddenly show up. I'm using paid Infura RPC already (and tested various ones in the past) but still seeing these issues from time to time.

**Comment by kiknaio** [comment]
I've updated Gas price based on this [article](https://cryptoslate.com/polygon-matic-to-raise-gas-fee-to-30-gwei-to-prevent-spam-transactions/#:~:text=Polygon%20%E2%80%BA%20DeFi-,Polygon%20(MATIC)%20to%20raise%20gas%20fee%20to%2030%20gwei%20to,order%20to%20combat%20spam%20transactions.) and now it works

```
gas: 9721975, // You should change it based on your requirements
gasPrice: 30000000000,
```

**Comment by dkillen** [comment]
Thanks @kiknaio Glad you got it working. @racetrack0616 and @PxGnome I suggest you try making this change to your configuration and try again.

**Comment by ThinkerTechnology** [comment]
UnhandledRejections detected
Promise {
  <rejected> {
    code: -32603,
    message: 'ESOCKETTIMEDOUT',
    data: { originalError: [Object] }
  }
} {
  code: -32603,
  message: 'ESOCKETTIMEDOUT',
  data: { originalError: { code: 'ESOCKETTIMEDOUT', connect: false } }
}


I am facing this issue while deploy smart contract on matic mainnet i have also try several RPCs but nothing work please guide me is there any issue with RPC of truffle
